src/main/java/io/cdap/http/internal/ForwardingEventExecutorGroup.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.http.internal;
+
+import io.netty.util.concurrent.EventExecutor;
+import io.netty.util.concurrent.EventExecutorGroup;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.ScheduledFuture;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * A {@link EventExecutorGroup} that forwards all methods to another {@link EventExecutorGroup}.
+ */
+public class ForwardingEventExecutorGroup implements EventExecutorGroup {
```

Comment:
Can we use `AbstractEventExecutorGroup` instead? It's forwarding to `next()`

Reply:
Are you suggesting to have `next()` returning the `delegate`? Then it is not exactly delegation since the `next()` method is returning `delegate` instead of `delegate.next()`?